File: app/src/main/java/com/banquemisr/data/remote/ApiClient.java
package com.banquemisr.data.remote;


import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {
    private static Retrofit retrofit;
    private static final String BASE_URL = "https://cafe-app-project.herokuapp.com/";



    public static Retrofit getRetrofit() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();


        }
        return retrofit;

    }

}

File: app/src/main/java/com/banquemisr/data/remote/ApiClient.kt
package com.banquemisr.data.remote

import com.banquemisr.coffeeapp_banquemisr.data.remote.Constants
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import okhttp3.OkHttpClient
import com.banquemisr.coffeeapp_banquemisr.domain.repositories.OAuthInterceptor

object ApiClient {

//    var retrofit: Retrofit? = null
//        get() {
//            if (field == null) {
//                field = Retrofit.Builder()
//                    .baseUrl(BASE_URL)
//                    .addConverterFactory(GsonConverterFactory.create())
//                    .build()
//            }
//            return field
//        }
//        private set
    private const val BASE_URL = "https://cafe-app-project.herokuapp.com/api/"
    val retrofitWithClient: Retrofit
        get() {
            val client = OkHttpClient.Builder()
                .addInterceptor(
                    OAuthInterceptor(
                        "Bearer",
                        Constants.TOKEN
                    )
                )
                .build()
            return Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .client(client)
                .build()
        }
}